shared/Microsoft.Extensions.StackTrace.Sources/StackFrame/StackTraceHelper.cs

```diff
@@ -240,5 +233,30 @@ private static bool TryResolveStateMachineMethod(ref MethodBase method, out Type
 
             return false;
         }
+
+        private static bool HasStackTraceHiddenAttribute(MemberInfo memberInfo)
+        {
+            IList<CustomAttributeData> attributes;
+            try
+            {
+                // Accessing MembmerInfo.GetCustomAttributesData throws for some types (such as types in dynamically generated assemblies).
+                // We'll skip looking up StackTraceHiddenAttributes on such types.
+                attributes = memberInfo.GetCustomAttributesData();
```

Comment:
@benaadams `MemberInfo.CustomAttributes` and `MemberInfo.GetCustomAttributesData` are seemingly equivalent. Would you know switching it over to use this correct?

Reply:
Hmm... What about
```csharp
foreach (var attibute in method.GetCustomAttributes(inherit: false))
{
    // internal Attribute, match on name
    if (attibute is CustomAttributeData cad && cad.AttributeType.Name == "StackTraceHiddenAttribute")
    {
        return false;
    }
}
```
Does that avoid the exception?

Reply:
Ahh... are they reflection Emit stuff?

Might need to change it in coreclr :-/

Reply:
Using `GetCustomAttributes` works in this scenario. However, we would likely still have to guard against `GetCustomAttributes` throwing because constructing an attribute might throw.